chatkit-core/src/test/kotlin/com/pusher/chatkit/rooms/RoomStoreApplyingReadStateUpdatedEventTest.kt

```diff
@@ -0,0 +1,50 @@
+package com.pusher.chatkit.rooms
+
+import com.google.common.truth.Truth.assertThat
+import com.pusher.chatkit.assertTypedSingletonList
+import com.pusher.chatkit.rooms.api.RoomMembershipApiType
+import com.pusher.chatkit.rooms.api.RoomReadStateApiType
+import com.pusher.chatkit.simpleRoom
+import com.pusher.chatkit.users.UserInternalEvent
+import com.pusher.chatkit.users.UserSubscriptionEvent
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+
+object RoomStoreApplyingReadStateUpdatedEventTest : Spek({
+
+    describe("given RoomStore with a Room with unreadCount of 0") {
+        val initialRoom = simpleRoom("roomId1", "General")
+        val subject by memoized {
+            RoomStore().apply {
+                initialiseContents(
+                        listOf(initialRoom),
+                        listOf(RoomMembershipApiType("roomId1", listOf("alice"))),
+                        listOf(RoomReadStateApiType("roomId1", 0, null))
+                )
```

Comment:
While I understand keeping test cases simple for understandability, I prefer tests which test a single isolated change in a slightly better populated fixture.

Code which took in this event and emitted an event for a random room from the store would still pass all the tests below, as would code which emitted events for *all* rooms in the store. Mixing a little irrelevant data in to the test fixture is a practical way to get a lot more coverage when testing collections.

You could also test the non-effect on other rooms explicitly, but that would still require additional rooms to exist.

Reply:
I tend to consider my test fixture data in terms of "what's the most wildly incorrect implementation which is not detectable by anything running in this fixture?"

Reply:
Good point! Addressed: d445eab6459ea8eff74c1ba8134caf13ce1a5739